### PR TITLE
MBS-9548: Filter external links by unique target

### PR DESCRIPTION
**Pull request by yvanzo**
### Fix [MBS-9548](https://tickets.metabrainz.org/browse/MBS-9548): Same link showing twice on the sidebar

Restore `unique` filtering on external links from late [root/layout/sidebar/sidebar-favicons.tt#L31](https://github.com/mwiencek/musicbrainz-server/blob/3205207c3c4f524e62119489957dc05531db162f/root/layout/sidebar/sidebar-favicons.tt#L31).

### CODE REVIEW

**Comment by yvanzo** [comment]
@mwiencek: What’s the most appropriate to test this regression? Selenium?

File: root/layout/components/ExternalLinks.js [hunk 1]
```diff
@@ -87,10 +87,18 @@ const ExternalLinks = ({entity, empty, heading}) => {
...
-    return compare(a.target.sidebar_name, b.target.sidebar_name);
+    return compare(a.target.sidebar_name, b.target.sidebar_name) ||
+      compare(a.target.href_url, b.target.href_url);
   });
 
-  links.push.apply(links, otherLinks.map(function (relationship) {
+  const uniqueOtherLinks = otherLinks.filter(
+    function (link, index, sortedLinks) {
+      return index === 0 ||
+        sortedLinks[index - 1].target.href_url !== link.target.href_url;
```
Comment (mwiencek) [inline root/layout/components/ExternalLinks.js]: Would `const uniqueOtherLinks = _.uniqBy(otherLinks, x => x. target.href_url)` (using `uniqBy` from lodash) also do the trick? I think that's what the sort-and-compare-pairwise logic does, but hard to be sure :)

**Comment by mwiencek** [comment]
Yes, I think I would add some test links to t/sql/selenium.sql and load the page using Selenium.

File: root/layout/components/ExternalLinks.js (hunk 1 above)
Comment (mwiencek) [reply root/layout/components/ExternalLinks.js]: Ah true. Maybe it can be sorted after the uniq instead of before, but probably not worth futzing with.

Comment (yvanzo) [reply root/layout/components/ExternalLinks.js]: I just found that Lodash provides [`sortedUniqBy`](https://lodash.com/docs#sortedUniqBy) too! I used it with your first suggestion in https://github.com/metabrainz/musicbrainz-server/pull/641/commits/4ad24ac319205b1df009fc62ad721922b961195.

**Pull request merged**